Framework/Core/include/Framework/AnalysisHelpers.h

```diff
@@ -348,7 +348,7 @@ struct IndexSparse {
 };
 
 /// This helper struct allows you to declare index tables to be created in a task
-template <typename T, typename IP = IndexSparse>
+template <typename T, typename IP = std::conditional_t<aod::MetadataTrait<T>::metadata::exclusive, IndexExclusive, IndexSparse>>
```

Comment:
Why do you need the IP in the template signature? Can't you simply use it below?

Reply:
Thanks. Looks like there is some other adjustments to be done following that.